src/Geta.NotFoundHandler.Admin/Areas/GetaNotFoundHandlerAdmin/Pages/Administer.cshtml.cs

```diff
@@ -110,20 +115,54 @@ public IActionResult OnPostImportRedirects()
             });
         }
 
-        var redirects = _redirectsXmlParser.LoadFromStream(ImportFile.OpenReadStream());
-
-        if (redirects.Any())
+        if (ImportFile.IsXml())
         {
-            _redirectsService.AddOrUpdate(redirects);
-            Message = $"{redirects.Count()} urls successfully imported.";
-            CardType = CardType.Success;
+            var redirects = _redirectsXmlParser.LoadFromStream(ImportFile.OpenReadStream());
```

Comment:
This part can be moved into a separate method where you will pass as a parameter interface IRedirectsParser, to clean a bit of code.

`private IActionResult ProcessFile(IRedirectsParser)`